kolibri/core/assets/src/core_app_constructor.js

```diff
@@ -24,10 +24,17 @@ const publicMethods = [
  * is configured in webpack.config.js
  * @classdesc
  */
-function Lib() {
+function Lib(kolibriGlobal) {
   // libraries
   this.loglevel = require('loglevel');
   this.vue = require('vue');
+  this.vue.mixin({
+    methods: {
+      url: (url_name, ...args) => {
+        return kolibriGlobal.urls[url_name](...args);
+      }
```

Comment:
> What are the scenarios in which we use this reverse functionality?

Referencing API endpoints by name instead of by url. For example `urls.deviceowner_list()` returns the url for the list endpoint for that resource. This corresponds exactly to Django's named urls. When a Django urlconf takes a positional argument, then that corresponds to the positional arguments of the js function. Named arguments are passed by passing an object in js.